packages/flutter_riverpod/test/framework_test.dart

```diff
@@ -47,6 +47,30 @@ void main() {
     );
   });
 
+  testWidgets('Supports multiple ProviderScopes in the same tree', (tester) async {
+    await tester.pumpWidget(
+      Column(
```

Comment:
There should always be a single root ProviderScope. Otherwise the branches don't share state.

Reply:
The original test used ProviderScopes, and you're right adding a root ProviderScope solved the issue.

In my storybook app I actually use UncontrolledProviderScopes with a root ProviderScope, and this still produces the error.

I've updated the app to use UncontrolledProviderScopes.

Just for context, my use case for this setup is a storybook style app where multiple instances of my app are rendered independently side by side.